core-it-suite/src/test/java/org/apache/maven/it/MavenITmng0553SettingsAuthzEncryptionTest.java

```diff
@@ -86,18 +93,13 @@ protected void setUp()
         handlerList.addHandler( repoHandler );
         handlerList.addHandler( new DefaultHandler() );
 
-        server = new Server( 0 );
         server.setHandler( handlerList );
         server.start();
-        while ( !server.isRunning() || !server.isStarted() )
+        if ( server.isFailed() )
```

Comment:
Server could have successfully started up, and then be successfully stopped, before this line hits.
In this scenario, the server LifeCycle state is STOPPED, not FAILED.

Reply:
because this has a purpose. I want to catch all errors. Not only IOException.
```
    @Override
    public final void start() throws Exception
    {
        synchronized (_lock)
        {
            try
            {
                if (_state == __STARTED || _state == __STARTING)
                    return;
                setStarting();
                doStart();
                setStarted();
            }
            catch (Throwable e)
            {
                setFailed(e);
                throw e;
            }
        }
    }
```

Reply:
That is the generic `start()` for `AbstractLifeCycle`.
`Server` is a `ContainerLifeCycle` and a special one at that.
It's `Server.doStart()` has to do far more then this abstract base behavior simply because it's a combination of a Container (with child lifecycles) and it's the main entry point for external that care about the LifeCycle state.

Reply:
I think the purpose of this is to check server's status in a parallel thread. ok, we can avoid the IF statement.

Reply:
If you have parallel processes sure.
But your usages are sequential.
If you decided to use something like DeploymentManager to scan for webapps and deploy them, then you would be interested in ThrowUnavailableOnStartup, which waits for initial deployment to report failures to Server.start